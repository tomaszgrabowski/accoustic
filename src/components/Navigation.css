nav {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin-bottom: 1rem;
}

.link {
    flex: 1;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    height: 1.8rem;
}

.link::after {
    position: absolute;
    top: 0;
    left: 0;
    display: inline-block;
    height: 100%;
    width: 100%;
    z-index: -1;
    content: '';
}

.link:hover {

}

.link:hover::after {
    top: 20px;
    border-bottom: 2px solid red;
}
