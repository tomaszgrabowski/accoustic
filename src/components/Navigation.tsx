import React from 'react';
import { NavLink } from 'react-router-dom';
import { ROUTES } from '../constants';

const Navigation = () => {
    return (
        <nav>
            <NavLink to={ ROUTES.HOME }>Główna</NavLink>
            <NavLink to={ ROUTES.ARTICLE }>Zadanie</NavLink>
            <NavLink to={ ROUTES.ABOUT_ME }>O mnie</NavLink>
        </nav>
    );
};

export default Navigation;
