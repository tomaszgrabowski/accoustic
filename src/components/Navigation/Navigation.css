nav {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin-bottom: 1rem;
}

.link {
    flex: 1;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    height: 1.8rem;
}

.link-active{
    border-bottom: 3px solid #333;
}
