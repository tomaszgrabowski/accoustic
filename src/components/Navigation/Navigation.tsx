import React from 'react';
import { NavLink } from 'react-router-dom';
import { ROUTES } from '../../constants';
import './Navigation.css';

const Navigation = () => {
    return (
        <nav className='nav'>
            <NavLink className='link' activeClassName='link-active' to={ ROUTES.HOME }>Home</NavLink>
            <NavLink className='link' activeClassName='link-active' to={ ROUTES.ARTICLE }>Task</NavLink>
            <NavLink className='link' activeClassName='link-active' to={ ROUTES.ABOUT_ME }>About me</NavLink>
        </nav>
    );
};

export default Navigation;
