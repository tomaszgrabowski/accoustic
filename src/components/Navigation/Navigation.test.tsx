import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navigation from './Navigation';

test( 'given Footer component when render then renders properly', () => {
    const wrapper = renderer.create( <BrowserRouter><Navigation/></BrowserRouter> ).toJSON();
    expect( wrapper ).toMatchSnapshot();
} );
