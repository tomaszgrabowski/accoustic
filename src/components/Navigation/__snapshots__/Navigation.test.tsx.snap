// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`given Footer component when render then renders properly 1`] = `
<nav
  className="nav"
>
  <a
    aria-current="page"
    className="link link-active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    Home
  </a>
  <a
    aria-current={null}
    className="link"
    href="/article"
    onClick={[Function]}
  >
    Task
  </a>
  <a
    aria-current={null}
    className="link"
    href="/aboutme"
    onClick={[Function]}
  >
    About me
  </a>
</nav>
`;
